(#8063) refactor some interfaces of storage layer column

wangbo opened the pull request:
# Proposed changes

Issue Number: close #8063

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

Commit:
(#8063) refactor some interfaces of storage layer column

HappenLee commented on be/src/vec/columns/column.h:
```diff
@@ -171,6 +171,19 @@ class IColumn : public COW<IColumn> {
+      LOG(FATAL) << "Method insert_many_dict_data is not supported for " << get_name();
+    }
+ 
+    virtual void insert_many_binary_data(size_t num, char* data_array, uint32_t* len_array, uint32_t* start_offset_array) {
```
```size_t num``` like other interface set last

HappenLee commented on be/src/olap/rowset/segment_v2/segment_iterator.cpp:
```diff
@@ -943,9 +939,9 @@ Status SegmentIterator::next_batch(vectorized::Block* block) {
 
         // 4.2 output short circuit predicate column
-        _output_column_by_sel_idx(block, _short_cir_pred_column_ids, sel_rowid_idx, selected_size, is_mem_reuse);
+        return _output_column_by_sel_idx(block, _short_cir_pred_column_ids, sel_rowid_idx, selected_size, is_mem_reuse);
```
here return the vec_pred_column will not exec

HappenLee commented on be/src/olap/rowset/segment_v2/binary_plain_page.h:
```diff
@@ -238,48 +238,15 @@ class BinaryPlainPageDecoder : public PageDecoder {
-                uint32_t len = offset(_cur_idx + 1) - start_offset;
-                dst_col_ptr->insert_data(&_data[start_offset], len);
-            }
+       uint32_t len_array[max_fetch];
```
format

HappenLee commented on be/src/vec/columns/column_vector.h:
```diff
@@ -212,11 +272,11 @@ class ColumnVector final : public COWHelper<ColumnVectorHelper, ColumnVector<T>>
     void insert_indices_from(const IColumn& src, const int* indices_begin, const int* indices_end) override;
 
-    void insert_elements(void* elements, size_t num) {
+    void batch_set_null_bitmap(uint8_t val, size_t num) {
```
look like same ase insert elements? why not use `insert_elements`

Commit:
1 format binary plain
2 remove batch_set_null_bitmap
3 fix segiter return value
4 set insert_many_binary_data args

HappenLee reviewed (approved):
LGTM
